sdk/python/feast/infra/offline_stores/contrib/spark_offline_store/spark_source.py

```diff
@@ -185,6 +185,18 @@ def get_table_query_string(self) -> str:
 
         return f"`{tmp_table_name}`"
 
+    # Note: Python requires redefining hash in child classes that override __eq__
+    def __hash__(self):
+
+        return super().__hash__()
```

Comment:
Should this also include `self.spark_options`?

Reply:
It doesn't necessarily need to as the two pieces of logic are used for different things. Take a look at the [redshift_source.py example](https://github.com/feast-dev/feast/blob/master/sdk/python/feast/infra/offline_stores/redshift_source.py#L116-L131) for how it is implemented elsewhere.